### DISCUSSION TITLE
How to gracefully end an experiment if participant does not consent

### DISCUSSION THREAD

**Discussion by dcousin3** [jspsych/jsPsych#3239] [Q&A]
I have a timeline with various events. The first is to show a consent form with two buttons ("consent" or "I do not consent"). When the second button is pressed, I would like the experiment to be redirected to the last event in the timeline (i.e., skip the intermediate events).

Is there a convenient way to achieve this? Most likely, this will use the `on_finish` attribute of the consent form, 

```
on_finish: function(data) {
           if (data.response == 1) { //does not consent
                 console.log("Quitting. Thanks!");
                 // what else to skip the next events and reach the last event?
           }
 }
```

**Comment by Shaobin-Jiang** [comment]
You can use [`jsPsych.endExperiment`](https://www.jspsych.org/7.3/reference/jspsych/#jspsychendexperiment). Or you could wrap the trials you would like to skip in a child timeline and add a `conditional_function` that returns `false` if the participant chooses "I do not consent".

**Comment by dcousin3** [comment]
Thank you, the second suggestion works like a charm.
